### DISCUSSION TITLE
Custom model using SDD scene

### DISCUSSION THREAD

**Discussion by DevDigitalCittius** [asus4/tf-lite-unity-sample#340] [General]
Hi

I'm having some troubles using my custom TFlite model, in the SDD scene.
I trained my model using this [google collab notebook ](https://colab.research.google.com/github/EdjeElectronics/TensorFlow-Lite-Object-Detection-on-Android-and-Raspberry-Pi/blob/master/Train_TFLite2_Object_Detction_Model.ipynb), and the result is kind of acceptable for the testing that im doing, as is shown in the image below.

![image](https://github.com/asus4/tf-lite-unity-sample/assets/151771961/22863266-b3c5-40b6-b6dd-0cc9e76650d3)

When i try to use it in the SDD scene, i get this error message.
![image](https://github.com/asus4/tf-lite-unity-sample/assets/151771961/55c57955-1eff-4327-accb-42f3448a6e57)

The only way i found to use my model in the SDD scene, is quantizing the model, when i quantize it, the model runs normally in the SDD scene, but there is no detection at all (look at image bellow)

![image](https://github.com/asus4/tf-lite-unity-sample/assets/151771961/1da881bb-cccc-440f-b2e4-af45b4433eea)

After a lot of trial and error, i found out, that when i quantize my model, it just stops working as expected.
Is there a way I can use my model without receiving this error message? Or a way I can quantize it without crashing my model?

 [Here is](https://drive.google.com/file/d/169QzfYmcWnyFnKtLm_h-1BsP-G5ibORQ/view?usp=sharing) the tflite file for the working model

Sorry if i got something wrong, it's my 1st time training a tensorflow model

Thanks in advance

**Comment by asus4** [comment]
Hi @DevDigitalCittius,

Change `BaseImagePredictor<sbyte>` to `BaseImagePredictor<float>`. This allows the use of non-quantized 32-bit float input.
https://github.com/asus4/tf-lite-unity-sample/blob/b7692d125295f238407b5644d981ac454af99cbc/Assets/Samples/SSD/SSD.cs#L9-L10

Then, the output order is a bit different from this old example. change the outputs like the following ↓
Please check out https://github.com/asus4/tf-lite-unity-sample/issues/312#issuecomment-1701376252 for more details.

Screen shot: ![Screenshot 2023-08-31 at 18 16 28](https://github.com/asus4/tf-lite-unity-sample/assets/357497/577fff40-28ec-405c-be6e-43504e7ca262)

**Reply by DevDigitalCittius** [reply]
Hi @asus4 
I made those changes and it's working!!
I think I just need to train my model a bit more.

Thanks for the quick repply
Best regards.

**Reply by asus4** [reply]
Good luck 👍

I plan to update this Object Detection example from SSD MobileNet v2 to a more recent model—maybe EfficiantDet Lite, which might be easier to train the custom model.

https://developers.google.com/mediapipe/solutions/vision/object_detector
